chronicle-front/src/app/services/media-retrieval.service.ts

```diff
@@ -1,41 +1,174 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { from, Observable } from 'rxjs';
+import { concatMap, map, switchMap } from 'rxjs/operators';
 import { Note } from '../models/Note';
+import { Tag } from '../models/Tag';
 import { Video } from '../models/Video';
 
 @Injectable({
   providedIn: 'root'
 })
 export class MediaRetrievalService {
 
-  constructor() { }
+  constructor(private httpClient: HttpClient) { }
 
+  v : Video = {
+    id : 1,
+    description : "Routing with Angular",
+    userId : "1",
+    url : "https://www.w3schools.com/html/mov_bbb.mp4",
+    tags : [{tagid: "1",name : "topic", value : "Angular"}]
+  };
+  b : Video = {
+    id : 2,
+    description : "Introduction to OOP",
+    userId : "1",
+    url : "https://www.w3schools.com/html/mov_bbb.mp4",
+    tags : [{tagid: "2",name : "topic", value : "Java"}]
+  };
 
-
-
-  getNotes(): Observable<Note[]> {
-
-  const n: Note = {
+  n : Note = {
     id : 1,
     description : 'Note',
     userId : '',
     url : 'url',
     tags : []};
-  const notes: Note[] = [n, n, n];
-  return from([notes]);
+
+
+  public getAllTags() : Observable<Tag[]>{
+    return this.httpClient.get('http://localhost:8080/myapp/videos/available-tags')
```

Comment:
Let's leverage the `environment` here instead of hard coding.

Reply:
added environment import and stored urls there for consolidated access